### ISSUE TITLE
import/no-named-default: named `default` binding inside `import type { ... }` is not flagged

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25613]
## Summary

oxlint's `import/no-named-default` rule does not flag a named `default` binding when it appears inside a TypeScript type-only import (`import type { default as X } from '...'`). The equivalent value import is flagged correctly, and both ESLint 7 and ESLint 9 (with eslint-plugin-import) flag the type-only form, so the type-only variant is silently skipped only by oxlint.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`repro.ts`:

```ts
import type {default as Quill, Delta} from 'quill'
export type T = Quill | Delta
```

`.oxlintrc.json` (force-enabling the rule, all other categories off):

```json
{
	"plugins": ["import"],
	"categories": {"correctness": "off"},
	"rules": {"import/no-named-default": "error"}
}
```

Run:

```
oxlint -c .oxlintrc.json --disable-nested-config repro.ts
```

## Expected

The rule reports a violation on the `default as Quill` binding. ESLint with eslint-plugin-import@2.31.0 reports:

```
1:25  error  Use default import syntax to import 'Quill'  import/no-named-default
```

Verified on ESLint 7.23.0 (@typescript-eslint/parser@6.21.0) and ESLint 9.39.4 (typescript-eslint@8.62.1 parser), both with eslint-plugin-import@2.31.0 — it is not parser-version specific.

## Actual

oxlint 1.78.0 reports nothing (exit 0). The named-default binding inside `import type { ... }` is silently skipped.

The rule is otherwise implemented and functional — the same config fires correctly on the value form (`control.ts`):

```ts
import {default as Quill} from 'quill'
console.log(Quill)
```

```
control.ts:1:9: error import(no-named-default): Replace default import with named import. help: Forbid named default exports.
```

So it specifically misses the type-only import variant. Real codebases use `import type {default as X, Y} from '...'` to pull a default type alongside named types; in an ESLint→oxlint migration this is a silent coverage loss.

## Cross-checked

| Engine | Versions | `repro.ts` (type-only) | `control.ts` (value) |
| --- | --- | --- | --- |
| ESLint 7 | eslint@7.23.0, @typescript-eslint/parser@6.21.0, eslint-plugin-import@2.31.0 | flagged at 1:25 | flagged at 1:20 |
| ESLint 9 | eslint@9.39.4, typescript-eslint@8.62.1 (parser), eslint-plugin-import@2.31.0 | flagged at 1:25 | flagged at 1:20 |
| oxlint | 1.78.0 | not flagged (exit 0) | flagged at 1:9 |
